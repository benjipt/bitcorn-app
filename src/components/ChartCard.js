import React, { Component } from 'react'
import { BarChart, Bar, Legend, Tooltip, XAxis, YAxis, ResponsiveContainer } from 'recharts'

export default class ChartCard extends Component {
    render() {
        return (
            <div className="container mt-5 p-4 border-black rounded card-custom">
                <div className="text-center">
                    <h5>Balance History</h5>
                </div>
                <hr></hr>
                <ResponsiveContainer width="100%" height={300}>
                    <BarChart data={ this.props.data }>
                        <XAxis dataKey="date" />
                        <YAxis />
                        <Tooltip />
                        <Legend />
                        <Bar dataKey="amount" fill="#3E92CC" />
                    </BarChart>
                </ResponsiveContainer>
            </div>
        )
    }
}
