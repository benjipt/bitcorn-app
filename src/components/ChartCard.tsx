import { Tooltip, XAxis, YAxis, ResponsiveContainer, LineChart, Line } from 'recharts'

interface ChartCardProps {
  data: []
}

export default function ChartCard({ data }: ChartCardProps) {
    return (
        <div className="container mt-5 p-4 border-black rounded card-custom" data-testid="ChartCard-1">
            <div className="text-center">
                <h5>Balance History</h5>
            </div>
            <hr></hr>
            <ResponsiveContainer width="100%" height={550}>
                <LineChart data={ data } className="linechart-style" >
                    <XAxis dataKey="date" />
                    <YAxis />
                    <Tooltip />
                    <Line type="monotone" dataKey="amount" stroke="#FF3366" strokeWidth={4} />
                </LineChart>
            </ResponsiveContainer>
        </div>
    )
}
