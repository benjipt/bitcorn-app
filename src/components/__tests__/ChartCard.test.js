import { render, cleanup, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ChartCard from '../ChartCard'

describe('ChartCard', () => {
    beforeEach(() => {
        render(<ChartCard/>)
    })

    afterEach(() => {
        cleanup()
    })

    test('should render ChartCard component', () => {
        screen.getByTestId('ChartCard-1')
    })

    test('matches snapshot', () => {
        const tree = renderer.create(<ChartCard/>).toJSON()
        expect(tree).toMatchSnapshot()
    })
})