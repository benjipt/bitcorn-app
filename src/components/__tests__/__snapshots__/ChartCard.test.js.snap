// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartCard matches snapshot 1`] = `
<div
  className="container mt-5 p-4 border-black rounded card-custom"
  data-testid="ChartCard-1"
>
  <div
    className="text-center"
  >
    <h5>
      Balance History
    </h5>
  </div>
  <hr />
  <div
    className="recharts-responsive-container"
    style={
      Object {
        "height": 550,
        "maxHeight": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "width": "100%",
      }
    }
  />
</div>
`;
